### PR TITLE
Create a podman interface so we can mock it

### PR DESCRIPTION
**Pull request by machacekondra**
This PR add podman interface, so we can mock it to properly test: https://github.com/jakub-dzon/k4e-device-worker/pull/54

### CODE REVIEW

File: internal/workload/podman/podman.go [hunk 1]
```diff
@@ -11,21 +11,32 @@ import (
 	api2 "github.com/jakub-dzon/k4e-device-worker/internal/workload/api"
 )
 
-type Podman struct {
+type Podman interface {
```
Comment (masayag) [inline internal/workload/podman/podman.go:14]: no need for `//go:generate mockgen...` ?

File: internal/workload/wrapper.go [hunk 2]
```diff
@@ -40,8 +40,8 @@ type WorkloadWrapper interface {
 
 // Workload manages the workload and its configuration on the device
 type Workload struct {
-	workloads         *podman.Podman
-	netfilter         network.Netfilter
+	workloads         podman.Podman
+	netfilter         *network.Netfilter
```
Comment (masayag) [inline internal/workload/wrapper.go:44]: this will be a pointer to an interface

Comment (machacekondra) [reply internal/workload/wrapper.go]: Thanks! Wrong rebase.

File: internal/workload/podman/podman.go (hunk 1 above)
Comment (machacekondra) [reply internal/workload/podman/podman.go:15]: Added, thanks.

**Review by masayag — APPROVED** [review]

**Pull request merged**